pilot/pkg/model/push_context.go

```diff
@@ -60,6 +58,9 @@ type PushContext struct {
 	privateDestRuleByHostByNamespace map[string]map[Hostname]*combinedDestinationRule
 	publicDestRuleHosts              []Hostname
 	publicDestRuleByHost             map[Hostname]*combinedDestinationRule
+
+	// sidecars for each namespace
+	sidecarsByNamespace map[string][]*SidecarScope
```

Comment:
At this point - it may be cleaner to add a 'namespaces map[string]NamespaceConfigs' and a new struct with all the 
'my namespace' data. Maybe in a separate PR we can evaluate this.

Reply:
Yes thats the plan. I wanted to get the basic piece working before I consolidate this data structure

Reply:
SG.